ui/app/search/search-ctrl.js

```diff
@@ -26,30 +26,22 @@
         var oldParams = mlSearch.getParams();
         var newParams = mlSearch.getCurrentParams();
 
-        // restore params, if we have cached results, and no params provided
-        if (model.search.results && _.isEqual( {}, newParams )) {
+        // capture URL params in mlSearch and ctrl model
+        mlSearch.fromParams().then(function() {
 
-          $location.search( oldParams );
```

Comment:
coming from demo details page, the "show all demos" will shift from details to an empty search (i.e. no ?q=oldsearch). given the condition above:

```
_.isEqual( {}, newParams )
```

this will execute and force the url to change. in effect, 2 searches were performed.